File: src/main/scala/io/koju/autopos/trade/purchase/service/PurchaseService.scala
package io.koju.autopos.trade.purchase.service

import io.koju.autopos.trade.purchase.domain.{Purchase, PurchaseLine}
import org.springframework.data.domain.{Page, Pageable}

trait PurchaseService {

  def getPurchases(pageable: Pageable): Page[Purchase]

  def createNewPurchase(): Purchase

  def updatePurchase(purchase: Purchase): Purchase

  def addPurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

  def updatePurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

  def deletePurchaseLine(purchaseLine: PurchaseLine): Unit

}

File: src/main/java/com/kapilkoju/autopos/service/PurchaseService.java
package com.kapilkoju.autopos.service;

import com.kapilkoju.autopos.domain.Purchase;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

/**
 * Service Interface for managing Purchase.
 */
public interface PurchaseService {

    /**
     * Save a purchase.
     *
     * @param purchase the entity to save
     * @return the persisted entity
     */
    Purchase save(Purchase purchase);

    /**
     *  Get all the purchases.
     *  
     *  @param pageable the pagination information
     *  @return the list of entities
     */
    Page<Purchase> findAll(Pageable pageable);

    /**
     *  Get the "id" purchase.
     *
     *  @param id the id of the entity
     *  @return the entity
     */
    Purchase findOne(Long id);

    /**
     *  Delete the "id" purchase.
     *
     *  @param id the id of the entity
     */
    void delete(Long id);
}
